packages/charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx

```diff
@@ -1,90 +1,96 @@
 import { IProcessedStyleSet } from 'office-ui-fabric-react/lib/Styling';
-
-import { Legend } from '../Legend/Legend';
-
-import { IMultiStackedBarChartProps, IMultiStackedBarChartStyles } from './MultiStackedBarChart.types';
-import { IDataPoint, ILegendDataItem } from './StackedBarChart.types';
-import { StackedBarChart } from './StackedBarChart';
+import { ILegend, Legends } from '@uifabric/charting';
+import {
+  IChartDataPoint,
+  IChartProps,
+  IMultiStackedBarChartProps,
+  IMultiStackedBarChartStyles,
+  StackedBarChart
+} from './index';
 
 const getClassNames = classNamesFunction<{}, IMultiStackedBarChartStyles>();
 
 export class MultiStackedBarChartBase extends React.Component<IMultiStackedBarChartProps, {}> {
   private _width: number;
   private _barHeight: number;
   private _renderLegend: boolean | undefined;
-  private _blankBarColor: string;
-  private _defaultPaletteColors: string[];
   private _classNames: IProcessedStyleSet<IMultiStackedBarChartStyles>;
 
   public render(): JSX.Element {
-    const { data, chartTitles, legendData, theme } = this.props;
-
+    const { data, theme, hideRatio } = this.props;
     this._adjustProps();
-
+    const legends = this._getLegendData(data!, hideRatio!);
+    this._renderLegend = data ? true : false;
+    const { palette } = theme!;
     return (
       <div className={this._classNames.root}>
-        {data.map((points: IDataPoint[], index: number) => {
-          const colors = this._getColorsForStackedBarChart(points);
-          if (points.length === 0) {
-            points.push({ x: '', y: 100 });
+        {data!.map((points: IChartProps, index: number) => {
+          let currentFlagShowRatio = hideRatio ? (hideRatio[index] !== undefined ? hideRatio[index] : false) : false;
+          if (points.chartData!.length === 0) {
+            points.chartData!.push({ legend: '', data: 100, color: palette.neutralTertiaryAlt });
+            currentFlagShowRatio = true;
           }
           return (
             <div key={index} className={this._classNames.items}>
               <StackedBarChart
                 barHeight={this._barHeight}
                 width={this._width}
                 theme={theme}
-                chartTitle={chartTitles[index]}
-                colors={colors}
                 data={points}
                 hideLegend={true}
-                hideNumberDisplay={true}
+                hideNumberDisplay={currentFlagShowRatio}
+                isMultiStackedBarChart={true}
               />
             </div>
           );
         })}
-        {this._renderLegend && <Legend renderData={legendData!} />}
+        {this._renderLegend && <Legends legends={legends} />}
       </div>
     );
   }
 
-  private _adjustProps = (): void => {
-    const { theme, className, styles, width, barHeight, legendData } = this.props;
-    const { palette } = theme!;
+  private _legendClicked(): void {
+    alert('Legend clicked');
```

Comment:
> alert('Legend clicked'); [](start = 3, length = 25)

do we need this alert. this does not seem to example

Reply:
removed it